### ISSUE TITLE
okd-service-01 "module_stdout": "/bin/sh: /usr/bin/python3: No such file or directory

### ISSUE THREAD

**Issue by bernalhd** [Kubeinit/kubeinit#143]
Hi! I'm getting this error, i'm not sure how solve it.
```
TASK [Configure the cluster service node] ************************************************************************************************************

TASK [../../roles/kubeinit_okd : update packages] ****************************************************************************************************
failed: [hypervisor-01 -> 10.0.0.100] (item=okd-service-01) => {"ansible_loop_var": "item", "changed": false, "item": "okd-service-01", "module_stderr": "Shared connection to 10.0.0.100 closed.\r\n", "module_stdout": "/bin/sh: /usr/bin/python3: No such file or directory\r\n", "msg": "The module failed to execute correctly, you probably need to set the interpreter.\nSee stdout/stderr for the exact error", "rc": 127}

PLAY RECAP *******************************************************************************************************************************************
hypervisor-01              : ok=58   changed=18   unreachable=0    failed=1    skipped=16   rescued=0    ignored=4   
```
In this try i changed the task "install services requirements" at the top of kubeinit/roles/kubeinit_okd/tasks/10_configure_service_nodes.yml and added python3 as kubeinit_okd_service_dependencies , but didn't work.

```
TASK [Configure the cluster service node] ************************************************************************************************************

TASK [../../roles/kubeinit_okd : install services requirements] **************************************************************************************
failed: [hypervisor-01 -> 10.0.0.100] (item=okd-service-01) => {"ansible_loop_var": "item", "changed": false, "item": "okd-service-01", "module_stderr": "Shared connection to 10.0.0.100 closed.\r\n", "module_stdout": "/bin/sh: /usr/bin/python3: No such file or directory\r\n", "msg": "The module failed to execute correctly, you probably need to set the interpreter.\nSee stdout/stderr for the exact error", "rc": 127}
```
```
root@okd-service-01 ~]# sudo rpm -q python3
package python3 is not installed
[root@okd-service-01 ~]# cat /etc/centos-release
CentOS Linux release 8.2.2004 (Core) 
[root@okd-service-01 ~]# 
```
```
user@user-desktop:~/git_projects/kubeinit$ rgrep kubeinit_okd_service_dependencies -A5 |head -n5
kubeinit/roles/kubeinit_okd/defaults/main.yml:kubeinit_okd_service_dependencies:
kubeinit/roles/kubeinit_okd/defaults/main.yml-  - python3
kubeinit/roles/kubeinit_okd/defaults/main.yml-  - haproxy
kubeinit/roles/kubeinit_okd/defaults/main.yml-  - httpd
kubeinit/roles/kubeinit_okd/defaults/main.yml-  - bind
```
![dependencies](https://user-images.githubusercontent.com/69559290/104664320-b02cf780-56ad-11eb-9a15-57e34e931599.jpg)

Any ideas ?

**Comment by ccamacho** [comment]
Hi @bernalhd, we were discussing about it on slack.

Note: Using Ubuntu hypervisor, I ran KubeInit OKD against it and it ran well until the task for update packages on okd-service-01:

```
failed: [hypervisor-01 -> 10.10.10.10] (item=okd-service-01) => {"ansible_loop_var": "item", "changed": false, "item": "okd-service-01", "module_stderr": "Shared connection to 10.10.10.10 closed.\r\n", "module_stdout": "/bin/sh: /usr/bin/python3: No such file or directory\r\n", "msg": "The module failed to execute correctly, you probably need to set the interpreter.\nSee stdout/stderr for the exact error", "rc": 127}
```

The okd-service-01 inventory by default is set to install CentOS 8 Core, which does not have python3 installed by default. It did get some form of python (via KubeInit?), but not in /usr/bin/python3, rather /usr/libexec/platform-python. I did not know that Ansible supported per-host interpreter settings, so I will go with that for a permanent solution. I worked around the issue by setting the inventory interpreter path to match CentOS settings, then symlinked it on the ubuntu host as a test, which worked. As it stands, unless the hypervisor and guests support the same interpreter path, the inventory will require per-host interpreter vars. (I think)

What distro are you using in the hypervisor? I can bet is different from CentOS :)
